### ISSUE TITLE
Make a grid of story states

### ISSUE THREAD

**Issue by glynnis** [MapStory/mapstory#273]
This story needs to be completed before #275

What states can a story be in with regards to being published, unpublished, draft, saved vs. unsaved changes, etc.

Prep these states, discuss with @jonpmarino and @glynnis so that Design for the header can be completed.

- [ ] Finalize with Marino what states a story can be in:
- - [ ] Can users unpublish/withdraw a story?
- - [ ] If a user chooses to make changes to a published story, what options do we present them? Unpublish the story while editing then republish vs. create a draft that will overwrite the published story when republished
- - [ ] Can users discard a draft (of a previously published story)

@jonpmarino, 

StoryLayers can be:
- **draft**: only viewable by the owner from his/her profile), or an administrator (this should be rare, perhaps as a preview when first uploaded, or if there is a problem. Low priority)
- **public**: viewable in the search by any visitor to mapstory.org (StoryLayers should be public by default)
- **community editable:**  any _registered_ user can contribute edits to the storylayer (a layer has to be public to be community editable)
- **not community editable**:  only the layer owner can contribute edits to the storylayer. A layer could be "not community editable" in draft or public mode.

MapStories can be:
- **Draft** - only viewable by the owner from his/her profile), or
- **Published** - viewable by any visitor to mapstory.org site 

NOTE - Returning a mapstory to 'draft' unpublishes the story until it is re-published
NOTE -  MapStories can only use public StoryLayers since, by definition, a layer has to be public in order to be viewed and shared in a published mapstory.

(It looks like when a MapStory is "returned to draft" it successfully is hidden from explore, but can still be accessed with a direct link)

(It would be nice if the user in composer was given more detail about what happens when "saving as a draft", as they are when a story is published)

**Comment by jonpmarino** [comment]
@glynnis in response to your questions, I'd say:
- [ ] yes, a user can unpublish a story
- [ ] If a user makes changes to a published story, to me it makes sense to unpublish while editing, and then republish when finished. this presumably would maintain the mapstory URL and story #...thus not messing with any external embeds of that story that have occurred.
- [ ] yes, user can discard any story they create on mapstory.org...whether its in draft or published form. 
Question...what is the difference between a draft and an unpublished story? are they not the same thing? or if not, does it make sense to think of them the same way? If so, we'd have 2 states: 1) Draft (or 'unpublished') and 2) Published Story. Published means its visible in the search. Draft means its visible for me on my profile with a clear DRAFT indication, and that's it.

**Comment by glynnis** [comment]
- Also consider: how do we communicate to the user that a draft or published story uses layers that have had updates? Do we let the user pin their story to a particular commit on a layer?

**Comment by glynnis** [comment]
![unnamed-2](https://cloud.githubusercontent.com/assets/1529366/23633202/698a22ec-028a-11e7-9816-93f8e9bd7b9b.jpg)
![unnamed](https://cloud.githubusercontent.com/assets/1529366/23633203/699fb6a2-028a-11e7-8aee-745542d6b305.jpg)

Notes from a whiteboard session a few months ago that I found while clearing off my phone :) stashing here in case it's useful when this gets picked up again.

**Comment by emilyashley** [comment]
Reminder to address save fail recovery in this workflow. 

There are many ways a save can fail (and I'm sure we'll keep finding new ones) We might want to address how to recover from a mapstory save failure in a way that keeps as much story, chapter, and layer data as possible

@Coop56 suggested we should only lose whatever it is we've changed since the last save if a save fails, not everything.

Address which of these states is the 'last save' to revert to? (for each available state)
